pkg/controller/kogitodataindex/kogitodataindex_controller.go

```diff
@@ -175,6 +174,26 @@ func (r *ReconcileKogitoDataIndex) onDeploymentCreate(cli *client.Client, deploy
 		log.Warnf("No container definition for service %s. Skipping applying custom Data Index deployment configuration", kogitoService.GetName())
 	}
 
+	dataIndex := kogitoService.(*appv1alpha1.KogitoDataIndex)
+	// set to the default.
+	persistenceProvider := infrastructure.DefaultDataIndexPersistence
+	if propValue, ok := dataIndex.Spec.Config[infrastructure.DataIndexPersistenceTypeProp]; ok {
+		foundPersistenceType := infrastructure.PersistenceProvider(propValue)
+		if _, ok := infrastructure.PersistenceProviders[foundPersistenceType]; ok {
+			persistenceProvider = infrastructure.PersistenceProviders[foundPersistenceType]
+		} else {
+			log.Warnf("the persistence type %s is invalid, valid are %s, defaulting to %s",
+				foundPersistenceType,
+				infrastructure.PersistenceProviders,
+				persistenceProvider)
+		}
+	} else {
+		log.Infof("the % property was not found, defaulting persistence provider to %s",
+			infrastructure.DataIndexPersistenceTypeProp, persistenceProvider)
+	}
+
+	log.Infof("injecting %s with the persistence provider %s.", infrastructure.DataIndexPersistenceEnv, persistenceProvider)
+	framework.SetEnvVar(infrastructure.DataIndexPersistenceEnv, persistenceProvider, &deployment.Spec.Template.Spec.Containers[0])
```

Comment:
Why do we need this env var? :thinking:

Reply:
i don't know what happened here, for some reason it reused my last changes. will update it tomorrow. just ignore it.

Reply:
oh crap, I messed up with the branch names.
moved to https://github.com/kiegroup/kogito-cloud-operator/pull/621